core/storage/topic/gae_models.py

```diff
@@ -150,7 +153,23 @@ def get_by_name(cls, topic_name):
         """
         return TopicModel.query().filter(
             cls.canonical_name == topic_name.lower()).filter(
-                cls.deleted == False).get() #pylint: disable=singleton-comparison
+                cls.deleted == False).get() # pylint: disable=singleton-comparison
+
+    @classmethod
+    def get_by_url_fragment(cls, url_fragment):
+        """Gets TopicModel by url_fragment. Returns None if the topic with
+        name url_fragment doesn't exist.
+
+        Args:
+            url_fragment: str. The url fragment of the topic.
+
+        Returns:
+            TopicModel|None. The topic model of the topic or None if not
+            found.
+        """
+        return TopicModel.query().filter(
```

Comment:
Please add a TODO and file an issue to find a way to make this faster. We should perhaps generate the full mapping once and store it in the memcache, or something. But doing a full query every time someone accesses a URL is kind of expensive.

Reply:
Done. Created #10210 to track this.